src/config/config.c

```diff
@@ -558,6 +577,51 @@ __config_process_value(WT_CONFIG *conf, WT_CONFIG_ITEM *value)
 		 */
 		if (value->type == WT_CONFIG_ITEM_NUM && errno == ERANGE)
 			goto range;
+	} else if (punescaped != NULL &&
+	    value->type == WT_CONFIG_ITEM_STRING &&
+	    __config_strnchr(value->str, '\\', value->len) != NULL) {
+		__wt_realloc(conf->session, NULL, value->len, punescaped);
```

Comment:
@ddanderson I've been sitting on this pull request because I'm very uncomfortable adding a memory allocation into this code path, but I haven't put in enough thought to make a concrete alternative suggestion. Can you think of another way around this problem?